spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -99,5 +99,47 @@ module.exports = function (provider, setup) {
       });
   });
 
+  it("Pause and Resume work", function (done) {
+    socket.connect('www.google.com', 80, function () {
+      var paused = false;
+      var messageCount = 0;
+      dispatch.on('onMessage', function(msg) {
+        if (!('data' in msg)) {
+          // Not an 'onData' message.
+          return;
+        }
+
+        // One onData is allowed during pause due to https://crbug.com/403076.
+        ++messageCount;
+        if (messageCount === 1) {
```

Comment:
this seems to expect 2 on data messages, 1 of which is expected to be during pause. If there's only a single onData message not delivered on pause would that be a problem?
I guess what I'm asking is if we're confident that this is general enough to fit the behavior of firefox/node as well.